Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift

```diff
@@ -2171,6 +2171,67 @@ class PaymentSheetDeferredServerSideUITests: PaymentSheetUITestCase {
         app.alerts.buttons["Confirm"].tap()
         XCTAssertTrue(app.staticTexts["Success!"].waitForExistence(timeout: 5.0))
     }
+
+    // MARK: - Multiple external payment methods/external Venmo
+    func testMultipleExternalPaymentMethodsPaymentSheet() throws {
+        var settings = PaymentSheetTestPlaygroundSettings.defaultValues()
+        settings.externalPaymentMethods = .all // test multiple external payment methods can load
+
+        loadPlayground(app, settings)
+
+        app.buttons["Present PaymentSheet"].waitForExistenceAndTap()
+
+        let payButton = app.buttons["Pay $50.99"]
+        // Assert PayPal is visible when using both external payment methods
+        XCTAssertTrue(scroll(collectionView: app.collectionViews.firstMatch, toFindCellWithId: "PayPal")?.waitForExistence(timeout: 5.0) ?? false)
+        guard let venmo = scroll(collectionView: app.collectionViews.firstMatch, toFindCellWithId: "Venmo") else {
+            XCTFail()
+            return
+        }
+        venmo.tap()
+        payButton.tap()
+        XCTAssertNotNil(app.staticTexts["Confirm external_venmo?"])
+        app.buttons["Cancel"].tap()
+
+        payButton.tap()
+        app.buttons["Fail"].tap()
+        XCTAssertTrue(app.staticTexts["Something went wrong!"].waitForExistence(timeout: 5.0))
+
+        payButton.tap()
+        app.buttons["Confirm"].tap()
+        XCTAssertTrue(app.staticTexts["Success!"].waitForExistence(timeout: 5.0))
+    }
+
+    func testMultipleExternalPaymentMethodsPaymentSheetFlowController() throws {
+        var settings = PaymentSheetTestPlaygroundSettings.defaultValues()
+        settings.externalPaymentMethods = .all // test multiple external payment methods can load
+        settings.uiStyle = .flowController
+
+        loadPlayground(app, settings)
+
+        app.buttons["Payment method"].waitForExistenceAndTap()
+        app.buttons["+ Add"].waitForExistenceAndTap()
+
+        // Assert PayPal is visible when using both external payment methods
+        XCTAssertTrue(scroll(collectionView: app.collectionViews.firstMatch, toFindCellWithId: "PayPal")?.waitForExistence(timeout: 5.0) ?? false)
+        scroll(collectionView: app.collectionViews.firstMatch, toFindCellWithId: "Venmo")?.waitForExistenceAndTap()
+
+        app.buttons["Continue"].tap()
```

Comment:
We might as well assert that the PaymentOptionDisplayData is correct, like check the image (maybe?) and label that flowcontroller vends.